### ISSUE TITLE
Run a user survey

### ISSUE THREAD

**Issue by adrn** [astropy/community-engagement#11]
At the NumFOCUS summit, we discussed the merits of running a short user survey to gather general information about user experience. For example, both [dask](dask/dask#4748) and [pandas](pandas-dev/pandas#27477) have recently run such surveys.

We have run surveys (either via google form or by mailing list post) in the past, but recently these have been for very specific goals (assessing performance, python 2 to 3, etc.). Others have also run more general surveys of the astronomical community about, e.g., [software use](https://arxiv.org/abs/1507.03989).

This issue is meant to start a discussion about whether we think it would be useful to run a similar survey. However, I think we should mainly discuss this face-to-face at the coordination meeting, and this issue should just serve as a place to remind us and collect broader feedback.

This is all open for discussion, but I was imagining a short (~5 min), information-gathering survey to assess how users use and perceive the astropy core package. For example, what operating systems they use, how they learn to use astropy, etc. To clarify, I'm not suggesting that we solicit general feedback about the package, but rather to gather more general statistics.

## Goals of a survey

I've included here a few goals for running such a survey, but please suggest modifications or additions.

* Gather statistics about the systems, computers, and environments in which people are using astropy
* Gather some very basic demographic statistics, e.g., at least country of residence/work, position (academic, industry, etc.), 
* Gather statistics about what subpackages our users use the most, and what other packages are being used with astropy most frequently
* Gather information about points of entry to astropy 
* Gather information about what educational materials people use, attendance to conferences (e.g., PyAstro, AstroHackWeek, AAS hack day, etc.)
* Gather information about perceptions of the developer community
* Understand whether our users know how to (if they want/need to): report a bug, get help/support, contribute code

## Open threads

* We should discuss or coordinate with NumFOCUS on how to review the questions. For example, is there a social scientist we can pay to consult on this?
* If we want to publish results (in a journal, e.g., BAAS), do we need IRB approval?
* Do we want to borrow questions from the [Python developers survey](https://www.jetbrains.com/research/python-developers-survey-2018/) to compare / benchmark against the broader Python community? Is that useful?

Discuss!

**Comment by kelle** [comment]
- I strongly think we need to have a clear idea about what we want this survey to do and what we would change about what we're doing based on the results. 
- I am more in favor of designing a survey which we could use to track longer term trends.

**Comment by adrn** [comment]
Agreed! I was imagining this being a yearly survey. I will write down some action items I was imagining and share for discussion.

**Comment by pllim** [comment]
As for motivation, maybe people who fill out the survey can be entered for a lucky draw to win something. :gift:

**Comment by adrn** [comment]
Returning to this, I'm still in favor of this idea! I think one sufficient motivation is just to gain a better understanding of the community of users that interface with the Astropy project -- I don't think we need to have a more specific goal. For example, Numpy last year ran a community survey with this goal: https://numpy.org/user-survey-2020/

**Comment by hamogu** [comment]
After reading over the numpy survey, I suggest to ask for specific feedback *at the same time*. I believe (but I might be wrong - the survey will show), that a sizable fraction of astropy users never provides feedback to us in the channels that we usually monitor (github issues, mailing list); are only rarely in more informal channels (facebook, twitter). Any opportunity to gather free-from feedback should be taken. We can take on the task of converting any specific feedback into github issues if warranted (based on experience with previous surveys, only a very low fraction of people will give such detailed feedback, but it's always worth trying).

**Comment by hamogu** [comment]
Just to note that down somewhere:
Specific questions I suggest:

- installation and pacakging: Do you use pip, conda, debian pacakges, ...?

**Comment by adrn** [comment]
Yes good idea! A quick update: I've been in touch with the Numpy leads who ran the survey and have some more information, so I will come up with a plan soon.

**Comment by adrn** [comment]
Note: the draft was never completed, but it is here: https://docs.google.com/document/d/1s1wbOmFjMQqaQC6yYCBBP1LNho0roZxU-GUtxqweq_A/edit?usp=sharing
The questions are based on the Numpy community survey. I was working to update the questions to be astropy-specific, and I was flagging questions that I thought could be removed to shorten the survey.

I still think this is an important thing to do, but at the time there wasn't much internal support for this so I lost momentum. If anyone wants to pick this up, I'd be happy to contribute or help work on this again.

**Comment by pllim** [comment]
Good talk about how to run survey from Scipy 2023: https://cfp.scipy.org/2023/talk/H9FDBV/

**Comment by pllim** [comment]
Yeah sorry, it is in the agenda but CoCo has too many things on the agenda right now.

**Comment by pllim** [comment]
> Gather statistics about the systems, computers, and environments in which people are using astropy

PyPI can probably give you an estimate already for this one.

**Comment by hamogu** [comment]
An idea maybe. But last time I looked, PyPi's and conda's download numbers were so large that is was obvious that they are totally dominated by CI - we don't have hundreds of human users installing astropy per day. Maybe there is a way to filter hat out today, but hearing for human users directly would be better.

**Comment by ceb8** [comment]
@adrn I'm so sorry we've sat on this so long and are now just coming back with questions.

1) Where is the new draft? I can only see the old one.
2) What kind of feedback do you want from the CoCo? Are you looking for an official sign off?

**Comment by jeffjennings** [comment]
Hey Clara, thanks to you all for coming back to this! the new draft is [here](https://docs.google.com/document/d/1Fzt49TpbfVwSpDMzPSCIt1gITeP09Yji1w8c7nt51wI/edit?usp=sharing). Feedback on the questions in the new draft of the survey, a sign-off, and thoughts on how to best distribute it would all be great to hear from the CoCo. Adrian might have more feedback in mind as well.

**Comment by ceb8** [comment]
@jeffjennings  OK! Thanks. We can provide that, hopefully on a shorter timescale this time.

**Comment by adrn** [comment]
@ceb8 Any updates on this?

**Comment by ceb8** [comment]
@adrn  You have the CoCo's official blessing.

**Comment by pllim** [comment]
CoCo6 has follow-up questions about the goals above given recent developments. While "blessing" is still granted, perhaps further discussions are needed before the survey is finalized.

**Comment by jeffjennings** [comment]
Adrian and I had the chance to discuss this, and we don't think it's an issue, as there aren't any aspects of the survey discussing or favoring initiatives in any relevant direction. There are a few standard demographic questions. And we think it's accurate that the work being done to complete and run the survey is all being done through a privately funded organization.
